app/components/work_packages/activities_tab/journals/index_component.rb

```diff
@@ -68,9 +68,12 @@ def base_journals
         end
 
         def fetch_journals
+          can_see_restricted = User.current.allowed_in_work_package?(:view_comments_with_restricted_visibility, work_package)
+
           API::V3::Activities::ActivityEagerLoadingWrapper.wrap(
             work_package
               .journals
+              .where(can_see_restricted ? nil : { restricted: false })
```

Comment:
👋🏾 I think a `visibile` active record scope would be more fitting, it would also remove the need for nil checks as scopes pass through AR relations 

Ref: https://github.com/search?q=repo%3Aopf%2Fopenproject%20visible&type=code

Reply:
Oh yeah, I'm still in the "get the thing to work" phase, before prettifying. But that's a good tip, will save me some time investigating. Thanks :bow: